src/config/schema.ts

```diff
@@ -0,0 +1,360 @@
+import { PackageMode } from '@opengovsg/formsg-sdk/dist/types'
+import awsInfo from 'aws-info'
+import convict, { Schema } from 'convict'
+import { email, url } from 'convict-format-with-validator'
+import { isNil } from 'lodash'
+import mongodbUri from 'mongodb-uri'
+import validator from 'validator'
+
+import {
+  Environment,
+  IBucketUrlSchema,
+  ICompulsoryVarsSchema,
+  IOptionalVarsSchema,
+  IProdOnlyVarsSchema,
+} from '../types'
+
+convict.addFormat(url)
+convict.addFormat(email)
+convict.addFormat({
+  name: 'string[]',
+  validate: (val: string[]) => {
+    if (!Array.isArray(val)) {
+      throw new Error('must be of type Array')
+    }
+    if (val.some((i) => typeof i !== 'string')) {
+      throw new Error('Elements must be of type string')
+    }
+  },
+  coerce: (val: string): string[] => {
+    return val.split(',')
+  },
+})
+
+/**
+ * Verifies that S3 bucket url is a valid url with or without trailing slash
+ */
+const validateBucketUrl = (
+  val: string,
+  {
+    isDev,
+    hasTrailingSlash,
+    region,
+  }: { isDev: boolean; hasTrailingSlash: boolean; region: string },
+) => {
+  if (!validator.isURL(val, { require_tld: !isDev })) {
+    throw new Error('must be a url')
+  }
+  if (hasTrailingSlash) {
+    if (!/[/]$/.test(val)) {
+      throw new Error('must end with a slash')
+    }
+  } else {
+    if (/[/]$/.test(val)) {
+      throw new Error('must not end with a slash')
+    }
+  }
+  // Region should be specified correctly in production
+  const isRegionCorrect = new RegExp(`^https://s3.${region}.amazonaws.com`, 'i')
+  if (!isDev && !isRegionCorrect.test(val)) {
+    throw new Error(`region should be ${region}`)
+  }
+}
+
+export const compulsoryVarsSchema: Schema<ICompulsoryVarsSchema> = {
+  awsConfig: {
+    imageS3Bucket: {
+      doc: 'S3 Bucket to upload images to',
+      format: String,
+      default: null,
+      env: 'IMAGE_S3_BUCKET',
+    },
+    logoS3Bucket: {
+      doc: 'S3 Bucket to upload logos to',
+      format: String,
+      default: null,
+      env: 'LOGO_S3_BUCKET',
+    },
+    attachmentS3Bucket: {
+      doc: 'S3 Bucket to upload encrypted attachments to',
+      format: String,
+      default: null,
+      env: 'ATTACHMENT_S3_BUCKET',
+    },
+  },
+  core: {
+    sessionSecret: {
+      doc: 'Session Secret',
```

Comment:
actually do we need a default for this in `docker-compose` as well? just trying to make sure breaking changes to the dev environment are minimal, since changing env vars can be very annoying especially for volunteers

Reply:
wow good catch. will add a default